src/sentry/static/sentry/app/components/events/interfaces/spans/scrollbarManager.tsx

```diff
@@ -0,0 +1,429 @@
+      // Find any span bar and adjust the virtual scroll bar's scroll position
+      // with respect to this span bar.
+
+      if (this.scrollableSpanBar.size === 0) {
+        return;
+      }
+
+      const spanBarDOM: HTMLDivElement = this.scrollableSpanBar.values().next().value;
+      this.syncVirtualScrollbar(spanBarDOM);
+    }
+
+    // If the window was selected via the minimap, then re-initialize the scroll state.
+
+    const viewWindowChanged =
+      prevProps.dragProps.viewWindowStart !== this.props.dragProps.viewWindowStart ||
+      prevProps.dragProps.viewWindowEnd !== this.props.dragProps.viewWindowEnd;
+
+    if (viewWindowChanged) {
+      this.initializeScrollState();
+    }
+  }
+
+  componentWillUnmount() {
+    this.cleanUpListeners();
+  }
+
+  scrollableSpanBar: Set<HTMLDivElement> = new Set();
+  contentSpanBar: Set<HTMLDivElement> = new Set();
+  virtualScrollbar: React.RefObject<HTMLDivElement> = React.createRef<HTMLDivElement>();
+  isDragging: boolean = false;
+  previousUserSelect: UserSelectValues | null = null;
+
+  initializeScrollState = () => {
+    // set initial scroll state for all scrollable spanbars
+    selectRefs(this.scrollableSpanBar, (spanBarDOM: HTMLDivElement) => {
+      spanBarDOM.scrollLeft = 0;
+    });
+
+    // reset all span bar content containers to their natural widths
+    selectRefs(this.contentSpanBar, (spanBarDOM: HTMLDivElement) => {
+      spanBarDOM.style.removeProperty('width');
+      spanBarDOM.style.removeProperty('max-width');
+    });
+
+    // Find the maximum content width. We set each content spanbar to be this maximum width,
+    // such that all content spanbar widths are uniform.
+    const maxContentWidth = Array.from(this.scrollableSpanBar).reduce(
+      (currentMaxWidth, currentSpanBar) => {
+        const isHidden = currentSpanBar.offsetParent === null;
+        if (!document.body.contains(currentSpanBar) || isHidden) {
+          return currentMaxWidth;
+        }
+
+        const maybeMaxWidth = currentSpanBar.scrollWidth;
+
+        if (maybeMaxWidth > currentMaxWidth) {
+          return maybeMaxWidth;
+        }
+
+        return currentMaxWidth;
+      },
+      0
+    );
+
+    selectRefs(this.contentSpanBar, (spanBarDOM: HTMLDivElement) => {
+      spanBarDOM.style.width = `${maxContentWidth}px`;
+      spanBarDOM.style.maxWidth = `${maxContentWidth}px`;
+    });
+
+    this.scrollableSpanBar.forEach(currentSpanBarRef => {
+      this.unregisterEventListeners(currentSpanBarRef);
+    });
+
+    const spanBarDOM: HTMLDivElement = this.scrollableSpanBar.values().next().value;
+
+    if (spanBarDOM) {
+      this.syncVirtualScrollbar(spanBarDOM);
+    }
+
+    this.scrollableSpanBar.forEach(currentSpanBarRef => {
+      this.registerEventListeners(currentSpanBarRef);
+    });
+  };
+
+  registerEventListeners = (spanbar: HTMLDivElement) => {
+    spanbar.onscroll = this.handleScroll.bind(this, spanbar);
```

Comment:
Does this need to be throttled or debounced to prevent jank? Should this be using `attachEventListener()` instead of the DOM1 interface?

Reply:
It's already throttled since `requestAnimationFrame` is used within `this.handleScroll`. Referenced here https://developer.mozilla.org/en-US/docs/Web/API/Document/scroll_event

I didn't use `attachEventListener()`, as that'll increase the complexity of having to track references of the attached event listeners. DOM0 event handlers can still be useful despite its age.